### ISSUE TITLE
person_id Generation in OMOP data

### ISSUE THREAD

**Issue by xinyuejohn** [USM-CHU-FGuyon/BlendedICU#15]
#### Issue Description

While working with MIMIC-III and MIMIC-IV datasets, I noticed that the transformation of `subject_id` to `person_id` in `preprocessed_labels.parquet` could lead to inconsistencies. The current method, as shown below, generates `person_id` based on the order of unique values in `person_source_value`:

```python
unique_pid = self.person['person_source_value'].unique()
start_idx = self.start_index['person']
self.person_id_mapper = pd.Series(range(start_idx, start_idx + len(unique_pid)), index=unique_pid)
self.person['person_id'] = self.person['person_source_value'].map(self.person_id_mapper)
```

#### Concerns

1. **Context-Specific Conversion**: This approach seems highly context-specific and potentially unstable across different tables or datasets. The unique generation of `person_id`s may not be preserved if tables are processed separately or at different times.

2. **Order Dependency**: The `person_id` mapping relies on the order of `unique_pid`, which can vary with different processing instances, leading to inconsistent `person_id` assignments.

3. **Data Linking Challenges**: For studies requiring linkage between different datasets (e.g., MIMIC-IV and MIMIC-CXR), this inconsistency in `person_id` generation could hinder the ability to effectively join datasets, especially when converted into the OMOP CDM format.

#### Suggested Alternative

To ensure consistency, I propose using a **hash function** or another traceable and consistent method for generating `person_id`s. This approach would guarantee that the same `person_source_value` is always mapped to the same `person_id`, irrespective of the dataset or processing instance.

#### Example Scenario

In my current project, I am attempting to integrate the MIMIC-CXR dataset (Chest X-ray dataset) into an OMOP dataset containing MIMIC-IV data (using your work). The inconsistency in `person_id` generation poses a significant challenge for this integration.

**Comment by USM-CHU-FGuyon** [comment]
Hi, 
So the issue lies in the creation of `person_id_mapper`: I should ensure that `person_source_value` always maps to the same `person_id`, no matter the input data... right ?

Does this seem reasonable to you ?

```
unique_pid = self.person.person_source_value.drop_duplicates()
pd.Series(unique_pid.apply(hash).values, index=unique_pid)


>> person_source_value
amsterdam-0       594850849731709773
amsterdam-1      5765600756235645567
amsterdam-10     4545262738506853931
amsterdam-89    -7809726421491983081
amsterdam-869    8070965104574124545
 ...
mimic3-42728    -2195033743443925061
mimic3-17532    -2494024862873726000
mimic3-13083    -3561420534028427962
mimic3-13620    -1134107389530107628
mimic3-7630     -6375162939243721547
```

I should probably apply that to `measurement_id` so the user can link an omop entry to the actual measurement in the original dataset, what do you think ?

**Comment by xinyuejohn** [comment]
Yes, this is exactly what I mean. There should be a robust and traceable method for mapping IDs during OMOP conversion. As of now, I am not aware of any other possibilities for linking on IDs besides using person_id. However, it wouldn't hurt to support this for potential future use cases.

**Comment by xinyuejohn** [comment]
Hi, I just ran the OMOP conversion (v0.2.2) and the type of person_id in person.csv is string. However, in OMOP CDM, it is supposed to be integer. Could you take a look at it?

**Comment by USM-CHU-FGuyon** [comment]
Hi, I should be able to fix this within a week, along with Issue #24

**Comment by USM-CHU-FGuyon** [comment]
`person_id` now  start with 1, and `visit_occurrence_id` start with 2, then I used 11 digits for encoding the original identifier. 


 visit_occurrence_id	| person_id
-----|---
 2367845152469	| 1367845152469
 2446728184613	| 1446728184613 
 2646177120648	| 1646177120648 
 2469987407475	| 1887476626291 
 2106057239409	| 1566595215392
